File: scripts/capture-screenshots.js
import { chromium } from 'playwright';
import { mkdirSync } from 'fs';
import { join } from 'path';

const PORT = 8765;
const BASE = `http://127.0.0.1:${PORT}`;

async function main() {
  mkdirSync(join('screenshots'), { recursive: true });

  const { spawn } = await import('child_process');
  const server = spawn('python3', ['-m', 'http.server', String(PORT)], {
    cwd: process.cwd(),
    stdio: 'ignore',
  });

  await new Promise(r => setTimeout(r, 800));

  const browser = await chromium.launch();
  const page = await browser.newPage({ viewport: { width: 1400, height: 900 } });

  await page.goto(BASE + '/index.html');
  await page.waitForTimeout(500);
  await page.screenshot({ path: 'screenshots/empty_state.png', fullPage: true });

  await page.click('#load-sample-btn');
  await page.waitForSelector('#dashboard', { state: 'visible', timeout: 10000 });
  await page.waitForTimeout(1200);
  await page.screenshot({ path: 'screenshots/loaded_state.png', fullPage: true });

  await browser.close();
  server.kill();
  console.log('Screenshots saved to screenshots/');
}

main().catch(e => {
  console.error(e);
  process.exit(1);
});

File: scripts/capture_screenshots.py
#!/usr/bin/env python3
import os
import subprocess
import sys
import time

ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
PORT = 8777

def main():
    try:
        from playwright.sync_api import sync_playwright
    except ImportError:
        print('pip install playwright && playwright install chromium')
        sys.exit(1)

    os.makedirs(os.path.join(ROOT, 'screenshots'), exist_ok=True)
    srv = subprocess.Popen([sys.executable, '-m', 'http.server', str(PORT)], cwd=ROOT, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
    time.sleep(0.8)

    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page(viewport={'width': 1400, 'height': 900})
        page.goto(f'http://127.0.0.1:{PORT}/index.html')
        page.wait_for_timeout(500)
        page.screenshot(path=os.path.join(ROOT, 'screenshots/empty_state.png'), full_page=True)
        page.click('#load-sample-btn')
        page.wait_for_selector('#dashboard', state='visible', timeout=10000)
        page.wait_for_timeout(1200)
        page.screenshot(path=os.path.join(ROOT, 'screenshots/loaded_state.png'), full_page=True)
        browser.close()

    srv.kill()
    print('Screenshots saved to screenshots/')


if __name__ == '__main__':
    main()
